packages/engine-test-utils/src/__tests__/engine-server/EngineEvents.spec.ts

```diff
@@ -0,0 +1,404 @@
+                  noteChangeEntries[0] as NoteChangeUpdateEntry;
+
+                expect(updatedEntry.status).toEqual("update");
+                expect(updatedEntry.note.fname).toEqual("foo");
+                expect(updatedEntry.note.stub).toBeTruthy();
+
+                done();
+              } catch (err) {
+                done(err);
+              }
+            }
+          );
+
+          engineClient.deleteNote(foo.id);
+        },
+        {
+          expect,
+          preSetupHook: ENGINE_HOOKS.setupBasic,
+          createEngine: createEngineFromServer,
+        }
+      );
+    });
+  });
+
+  describe("WHEN deleting a note that has no children", () => {
+    test("THEN delete event fired with the correct note props AND update event fired for its parent", async (done) => {
+      await runEngineTestV5(
+        async ({ engine }) => {
+          const engineClient = engine as DendronEngineClient;
+
+          // bar has no children in the ENGINE_HOOKS_MULTI.setupBasicMulti setup.
+          const bar = engine.notes["bar"];
+
+          engineClient.onEngineNoteStateChanged(
+            (noteChangeEntries: NoteChangeEntry[]) => {
+              try {
+                expect(noteChangeEntries.length).toEqual(2);
+
+                noteChangeEntries.forEach((entry) => {
+                  if (entry.status === "delete") {
+                    expect(entry.status).toEqual("delete");
+                    expect(entry.note.fname).toEqual(bar.fname);
+                  } else if (entry.status === "update") {
+                    const updateEntry = entry as NoteChangeUpdateEntry;
+
+                    expect(updateEntry.note.fname).toEqual("root");
+                    expect(updateEntry.note.children.length).toEqual(1);
+                    expect(updateEntry.prevNote.children.length).toEqual(2);
+                  } else {
+                    done({
+                      message: "Unexpectedly received a create NoteChangeEntry",
+                    });
+                  }
+                });
+
+                done();
+              } catch (err) {
+                done(err);
+              }
+            }
+          );
+
+          engineClient.deleteNote(bar.id);
+        },
+        {
+          expect,
+          preSetupHook: ENGINE_HOOKS.setupBasic,
+          createEngine: createEngineFromServer,
+        }
+      );
+    });
+  });
+
+  /**
+   * If test fails due to timeout, then it means at least one of the expected
+   * onNoteCreated/onNoteDeleted callbacks was not received.
+   *
+   * TODO: Current storeV2 behavior is to delete and then create a note during a
+   * rename (because the file system name has been changed). Instead of
+   * reporting a create + delete, should we actually just report a single
+   * update, as the ID doesn't change on the note? Furthermore, an update is
+   * reported on the Root node, even though prev/new states are identical. This
+   * bug should be fixsed.
```

Comment:
nitpick: typo

Reply:
for these comments, can we also create a task note and add it in the comment?

Reply:
I'll remove the part about create+delete vs update, as it looks like we need that for now for fileWatcher to properly work.